plugins/main/public/components/overview/vulnerabilities/searchbar/use-seach-bar-configuration.tsx

```diff
@@ -0,0 +1,128 @@
+import React, { useEffect, useState } from 'react'
+import { SearchBarProps, FilterManager, TimeRange, Query } from '../../../../../../../src/plugins/data/public'
+import { Filter, IIndexPattern, IndexPatternsContract } from '../../../../../../../src/plugins/data/public';
+import { getPlugins, getDataPlugin } from '../../../../kibana-services';
+
+import { useFilterManager, useQueryManager, useTimeFilter } from '../../../common/hooks';
+import { AUTHORIZED_AGENTS } from '../../../../../common/constants';
+
+// Input - types
+type tUseSearchBarCustomInputs = {
+    defaultIndexPatternID?: IIndexPattern['id'];
+    onFiltersUpdated?: (filters: Filter[]) => void;
+    onQuerySubmitted?: (payload: { dateRange: TimeRange; query?: Query }, isUpdate?: boolean) => void;
+}
+type tUseSearchBarProps =  Partial<SearchBarProps> & tUseSearchBarCustomInputs;
+
+// Output types
+type tUserSearchBarResponse = {
+    searchBarProps: Partial<SearchBarProps>;
+}
+
+/**
+ * Hook used to compose the searchbar configuration props
+ * @param props 
+ * @returns 
+ */
+const useSearchBarConfiguration = (props?: tUseSearchBarProps): tUserSearchBarResponse => {
+    // dependencies
+    const filterManager = useFilterManager().filterManager as FilterManager;
+    const [query, setQuery] = props?.query ? useState(props?.query) : useQueryManager();
+    const { timeFilter, timeHistory, setTimeFilter } = useTimeFilter();
+    // states
+    const [isLoading, setIsLoading] = useState(false);
+    const [indexPatternSelected, setIndexPatternSelected] = useState<IIndexPattern>();
+    const [error, setError] = useState();
```

Comment:
This state is not being used. Right?

Reply:
Yes, fixed in https://github.com/wazuh/wazuh-kibana-app/pull/5960/commits/2348731fb1410f74abdaeee6920585942d2e2c0a